File: C (CLang)/INF2SEM2/bubblesort.c
#include <stdio.h>

int main() {
    int v[100] = {6, 4, 8, 3, 1, 9, 5, 2, 9, 7};
    int schimbare, aux, lungime = 10, perete = 1;
    do {
        schimbare = 0;
        for (int i = 0; i < lungime-perete; i++) {
            if (v[i] > v[i + 1]) {
                aux = v[i];
                v[i] = v[i + 1];
                v[i + 1] = aux;
                schimbare++;
            }
        }
        perete++;

    } while (schimbare != 0);
    for (int i = 0; i < lungime; i++) {
        printf("%d ", v[i]);
    }

    return 0;
}

File: Facultate/python/teme/bubble_sort.py
def bbs():
    schimbare = True
    c = 0
    while schimbare:
        schimbare = False
        c += 1
        for i in range(len(lista) - c):
            if lista[i] > lista[i + 1]:
                schimbare = True
                aux = lista[i]
                lista[i] = lista[i + 1]
                lista[i + 1] = aux
            print(lista, c)

    print(lista)


if __name__ == "__main__":
    lista = [2, 5, 3, 9, 6, 4, 2, 1]
    bbs()
